sound/soc/intel/boards/sof_sdw.c

```diff
@@ -187,11 +187,12 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
 					SOF_RT715_DAI_ID_FIX |
 					SOF_SDW_FOUR_SPK),
 	},
+	/* AlderLake devices */
 	{
 		.callback = sof_sdw_quirk_cb,
 		.matches = {
-			DMI_MATCH(DMI_SYS_VENDOR, "Intel"),
-			DMI_MATCH(DMI_PRODUCT_NAME, "adlrvp"),
```

Comment:
On ADL RVP SDW platform, we will use rt714 for dmic, not PCH DMIC? If so, we should remove SOF_SDW_PCH_DMIC in the driver_data in the UEFI item.
On Chrome RVP, it will use RT711 + PCH DMIC if I understand correctly. If so, I think we may need to separate ADL UEFI (RVP) and coreboot (Chrome RVP).

Reply:
@libinyang yes, Chrome uses PCH DMIC.

Reply:
@plbossart sorry for delay, it was holiday in INDIA and hence didnt look at your previous message. 
with libin's comment, should we split the 2 match ?

Reply:
Hmm. I know coreboot upstream has adlrvp supported, so a case can be made to have the same in upstream kernel, but this is starting to get really messy.  I think we either not have adlrvp support in upstream at all, or maintain two entries and use some DMI field to identify coreboot. We coukd use DMI_BIOS_VENDOR -- it's "coreboot". And I'd keep "adlrvp" as the product name as that's the motherboard name in coreboot upstream (or did this change already go to coreboot).

Reply:
oh  man, this is getting complicated.

for ADL-SDL, the only SoundWire device is RT711, so we probably want to enable the DMIC.

But when we connect the 3-in-1 board, we will NOT use the RT711 on board and will test with the RT711-sdca, RT1316x2 and RT714-SDCA. so for this case we do not want the DMIC.

Likewise for ADL-RVP with whatever ends up being used for Chrome, we need to have DMICs enabled.

I think we need a generic rule which removes DMIC support when RT714 is used.

Reply:
@kv2019i the coreboot patch is still not landed. i'm ok with your suggestion, but will wait for @plbossart to confirm. 

If we want to have the split the DMI match, we can ask coreboot team to drop the change.

Reply:
splitting the match does not help, we have two configurations for the UEFI case.

Reply:
@plbossart so lets go with this fix. will talk to coreboot team to get the patch merged.
w cna have default as Rt711 + PCH DMIC . for using RT714 , add additional match i guess

Reply:
I added a patch to ignore the DMICs when we find an RT714/715. That should avoid issues when adding support for the gen3 3-in-1 board.